index.js

```diff
@@ -0,0 +1,125 @@
+const URL = 'https://trektravel.herokuapp.com/trips';
+
+const reportStatus = (message) => {
+  $('#status-message').html(message);
+};
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p>`
+  content += "<ul>";
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      content += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  content += "</ul>";
+  reportStatus(content);
+};
+
+const loadTrips = () => {
+  reportStatus('Loading trips...');
+  
+  axios.get(URL)
+    .then((response) => {
+      reportStatus(`Successfully loaded ${response.data.length} trips`);
+      const trips = response.data;
+      displayTripList(trips);
+    })
+    .catch((error) => {
+      reportStatus(`Encountered an error while loading trips: ${error.message}`);
+      console.log(error);
+    });
+};
+
+const displayTripList = (tripList) => {
+  $(".card").show();
+  const target = $('#trip-list');
+  target.empty();
+  tripList.forEach((trip) => {
+    target.append(`<a href="#" id="${trip.id}" class="list-group-item list-group-item-action">${trip.name}</a>`);
+    const getDetails = showTripDetails(trip)
+    $(`#${trip.id}`).click(getDetails);
+  });
+}
+
+const showTripDetails = (trip) => {
+  console.log("showing details for trip", trip.name);
+
+  const getTripDetails = () => {
```

Comment:
Good use of a closure here to keep track of the trip data.

Style note: when we use a closure in this way, we typically organize it as follows:

```js
const displayTripList = () => {
  // ...
  // The "outer function" is our forEach, defined anonymously
  tripList.forEach((trip) => {
    // The "inner function" (our closure) is also an anonymous function. 
    // All it does is call the function we really want (showTripDetails) with the right data.
    const getDetails = () => showTripDetails(trip);
    $(`#${trip.id}`).click(getDetails);
  }
}

// the worker function is written as though it weren't a closure
const showTripDetails = (trip) => {
  axios.get(URL + '/' + trip.id)
  // ...
}
```